contracts/wrappers/QuestPartner.sol

```diff
@@ -0,0 +1,356 @@
+
+    }
+
+    function emergencyWithdraw(uint256 questID, address recipient) external notKilled nonReentrant {
+        if(msg.sender != creators[questID]) revert Errors.CallerNotAllowed();
+        
+        board.emergencyWithdraw(questID, recipient);
+    }
+
+    function _pullTokens(address token, address source, uint256 amount) internal returns(bool){
+        IERC20(token).safeTransferFrom(source, address(this), amount);
+
+        IERC20(token).safeIncreaseAllowance(address(board), amount);
+
+        return true;
+    }
+
+    function _sendPartnerShare(address token, uint256 feesAmount) internal returns(bool){
+        uint256 partnerAmount = (feesAmount * partnerShare) / MAX_BPS;
+
+        chest.transferERC20(token, feesReceiver, partnerAmount);
+
+        return true;
+    }
+
+    function execute(
+        address to,
+        uint256 value,
+        bytes calldata data
+    ) external payable onlyAllowed notKilled returns (bool, bytes memory) {
+        // Since this contract is approved as a manager for the QuestTreasuryChest contract
+        // we do not want this method to be allowed to call the contract.
+        // And not allowed to call the Quest Board either, since we already have methods for that
+        if(to == address(board) || to == address(chest)) revert Errors.ForbiddenCall();
+
+        (bool success, bytes memory result) = to.call{value: value}(data);
+        require(success, _getRevertMsg(result));
+
+        return (success, result);
+    }
+
+    function _getRevertMsg(bytes memory _returnData)
+        internal
+        pure
+        returns (string memory)
+    {
+        if (_returnData.length < 68) return "Transaction reverted silently";
+
+        assembly {
+            _returnData := add(_returnData, 0x04)
+        }
+
+        return abi.decode(_returnData, (string));
+    }
+
+
+    // Partner methods
+    function recoverERC20(address token) external onlyAllowed returns(bool) {
+        uint256 amount = IERC20(token).balanceOf(address(this));
+        if(amount == 0) revert Errors.NullAmount();
+        IERC20(token).safeTransfer(owner(), amount);
+
+        return true;
+    }
+
+    function addVoterBlacklist(address account) external onlyAllowed notKilled returns(bool) {
+        //We don't want to have 2x the same address in the list
+        address[] memory _list = voterBlacklist;
+        uint256 length = _list.length;
+        for(uint256 i = 0; i < length;){
+            if(_list[i] == account){
+                return false;
+            }
+            unchecked {
+                ++i;
+            }
+        }
+
+        voterBlacklist.push(account);
+
+        emit AddVoterBlacklist(account);
+
+        return true;
+    }
+
+    function removeVoterBlacklist(address account) external onlyAllowed notKilled returns(bool) {
+        address[] memory _list = voterBlacklist;
+        uint256 length = _list.length;
+
+        for(uint256 i = 0; i < length;){
```

Comment:
map also here ?
we can have a map => int
if map[voter] = 1 => blacklisted
if map[voter] = 0 => ok

Reply:
Same as: https://github.com/PaladinFinance/Warden-Quest/pull/19#discussion_r885072075